File: src/game/object/Theme.java
package game.object;

import engine.Texture;
import game.Assets;

abstract public class Theme {
	abstract public Texture tileTexture();
	abstract public Texture backgroundTexture();
	abstract public Texture dangerTexture();

	public Theme() {}

	public static Theme PORTAL_THEME = new Theme() {
		@Override
		public Texture tileTexture() {
			return Assets.brickTexture;
		}

		@Override
		public Texture backgroundTexture() {
			return Assets.panelTexture;
		}

		@Override
		public Texture dangerTexture() {
			return Assets.lavaTexture;
		}
	};

	public static Theme HELL_THEME = new Theme() {
		@Override
		public Texture tileTexture() {
			return Assets.hellGroundTexture;
		}

		@Override
		public Texture backgroundTexture() {
			return Assets.hellBackgroundTexture;
		}

		@Override
		public Texture dangerTexture() {
			return Assets.lavaTexture;
		}
	};
}

File: src/game/object/Theme.kt
package game.`object`

import engine.Texture
import game.Assets

abstract class Theme {
	abstract fun tileTexture(): Texture
	abstract fun backgroundTexture(): Texture
	abstract fun dangerTexture(): Texture

	companion object {
		@JvmField
		var PORTAL_THEME: Theme = object : Theme() {
			override fun tileTexture(): Texture {
				return Assets.brickTexture
			}

			override fun backgroundTexture(): Texture {
				return Assets.panelTexture
			}

			override fun dangerTexture(): Texture {
				return Assets.lavaTexture
			}
		}
		@JvmField
		var HELL_THEME: Theme = object : Theme() {
			override fun tileTexture(): Texture {
				return Assets.hellGroundTexture
			}

			override fun backgroundTexture(): Texture {
				return Assets.hellBackgroundTexture
			}

			override fun dangerTexture(): Texture {
				return Assets.lavaTexture
			}
		}
	}
}
